### PR TITLE
Adding some extra CI things

### PR DESCRIPTION
**Pull request by geeksareforlife**

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -32,3 +34,59 @@ jobs:
...
+        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-php-
+
+    - name: Install dependencies
+      if: steps.composer-cache.outputs.cache-hit != 'true'
+      run: composer install --prefer-dist --no-progress --no-suggest
+
+    - name: Run test suite
+      run: composer run-script analyse
```
Comment (Brunty) [inline .github/workflows/php.yml:64]: With composer scripts, you don't need the `run-script` part, you can run them with just `composer analyse`

**Comment by geeksareforlife** [comment]
I can;t find out how to enforce the ordering of use statements.  I know that isn't in PSR-12, but I do like it!

**Comment by Brunty** [comment]
@geeksareforlife that's a part of the https://github.com/slevomat/coding-standard package:

```
    <rule ref="SlevomatCodingStandard.Namespaces.AlphabeticallySortedUses">
        <properties>
            <property name="caseSensitive" value="1"/>
        </properties>
    </rule>
```

**Comment by geeksareforlife** [comment]
@Brunty ah, I was going for sorting by length :) should I be sorting alphabetically?

**Comment by Brunty** [comment]
@geeksareforlife you can sort however you'd like, I tend to like alphabetically as it can make them easier to find in a list, but it's up to you! :)

**Comment by geeksareforlife** [comment]
Well, seeing as there doesn't seem to be any custom sniffs for length based rules, I will go alpha for now and have a proper think once it is all setup!

**Pull request merged**